src/lib/swapper/swappers/OsmosisSwapper/endpoints.ts

```diff
@@ -0,0 +1,210 @@
+      })
+
+      return unsignedTx
+    }
+
+    // TODO(gomes): uncomment me when actually implementing multi-hop
+    // const osmoAddress = sellAssetIsOnOsmosisNetwork ? sellAddress : receiveAddress
+    // const cosmosAddress = sellAssetIsOnOsmosisNetwork ? receiveAddress : sellAddress
+
+    /** At the current time, only OSMO<->ATOM swaps are supported, so this is fine.
+     * In the future, as more Osmosis network assets are added, the buy asset should
+     * be used as the fee asset automatically. See the whitelist of supported fee assets here:
+     * https://github.com/osmosis-labs/osmosis/blob/04026675f75ca065fb89f965ab2d33c9840c965a/app/upgrades/v5/whitelist_feetokens.go
+     */
+
+    const getFeeDataInput: Partial<GetFeeDataInput<OsmosisSupportedChainId>> = {}
+    const swapFeeData = await (sellAssetIsOnOsmosisNetwork
+      ? osmosisAdapter.getFeeData(getFeeDataInput)
+      : cosmosAdapter.getFeeData(getFeeDataInput))
+
+    const feeDenom = sellAssetIsOnOsmosisNetwork
+      ? 'uosmo'
+      : atomOnOsmosisAssetId.split('/')[1].replace(/:/g, '/')
+
+    // The actual amount that will end up on the IBC channel is the sell amount minus the fee for the IBC transfer Tx
+    // We need to deduct the fees from the initial amount in case we're dealing with an IBC transfer + swap flow
+    // or else, this will break for swaps to an Osmosis address that doesn't yet have ATOM
+    const sellAmountAfterFeesCryptoBaseUnit = sellAssetIsOnOsmosisNetwork
+      ? sellAmountBeforeFeesCryptoBaseUnit
+      : bnOrZero(sellAmountBeforeFeesCryptoBaseUnit).minus(swapFeeData.fast.txFee).toString()
+
+    // TODO(gomes): this is temporary while we plumb things through, this won't work for cross-account trades
+    // We shouldn't have a notion of an accountNumber once we hit the new Osmosis swapper endpoints
+    // or rather we should have it both on the sell and buy side?
+
+    const receiveAccountNumber = accountNumber
+
+    const osmoAddress = sellAssetIsOnOsmosisNetwork ? sellAddress : receiveAddress
+
+    const txToSign = await buildApiTradeTx({
+      osmoAddress,
+      accountNumber: sellAssetIsOnOsmosisNetwork ? accountNumber : receiveAccountNumber,
+      adapter: osmosisAdapter,
+      buyAssetDenom,
+      sellAssetDenom,
+      sellAmount: sellAmountAfterFeesCryptoBaseUnit,
+      gas: swapFeeData.fast.chainSpecific.gasLimit,
+      feeAmount: swapFeeData.fast.txFee,
+      feeDenom,
+    })
+
+    return txToSign
+  },
+
+  checkTradeStatus: async ({
+    txHash,
+    chainId,
+  }): Promise<{ status: TxStatus; buyTxHash: string | undefined; message: string | undefined }> => {
+    try {
+      const { REACT_APP_OSMOSIS_NODE_URL: osmoUrl, REACT_APP_COSMOS_NODE_URL: cosmosUrl } =
+        getConfig()
+      const chainAdapterManager = getChainAdapterManager()
+      const adapter = chainAdapterManager.get(chainId) as CosmosSdkChainAdapter | undefined
+      if (!adapter) throw new Error('Failed to get adapter')
```

Comment:
yeah an assetGetChainAdapter would be handy here too

Reply:
https://github.com/shapeshift/web/pull/4905/commits/62703de9e849401f457894cdb3db00f9f30a6a11